pkg/cmd/controller/controller_build.go

```diff
@@ -648,47 +648,53 @@ func (o *ControllerBuildOptions) updatePipelineActivityForRun(kubeClient kuberne
 	if allCompleted {
 		if failed {
 			spec.Status = v1.ActivityStatusTypeFailed
+		} else if pri.Type == tekton.MetaPipeline {
```

Comment:
I am not so familiar with this controller yet, but by looking at the code I am a bit confused by seeing something like `if allCompleted` and then  `ActivityStatusTypePending` if the type is tekton.MetaPipeline. Not quite sure how to do it differently atm though.

Reply:
would it make more sense to do the switch between the type once at the beginning of the function and handle the case of meta pipeline or execution pipeline?

Reply:
`allCompleted` refers to the current executing `PipelineRun`, not the `PipelineActivity` as a whole. We want to treat the meta and execution `PipelineRun`s both as part of the same `PipelineActivity`, so if the meta pipeline didn't fail, we're expecting there to be an execution pipeline to come that we'll add to the `PipelineActivity`.

Reply:
Sure. So it is more a "allStageCompleted"

Reply:
Yeah, `allStagesCompleted` feels right. I'll rename it.